storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriteHandler.java

```diff
@@ -130,15 +130,17 @@ public synchronized void write(Collection<ShufflePartitionedBlock> shuffleBlocks
     String dataFileName = ShuffleStorageUtils.generateDataFileName(fileNamePrefix);
     String indexFileName = ShuffleStorageUtils.generateIndexFileName(fileNamePrefix);
 
-    try (LocalFileWriter dataWriter = createWriter(dataFileName, dataBufferSize);
-        LocalFileWriter indexWriter = createWriter(indexFileName, indexBufferSize); ) {
+    try (FileWriter dataWriter =
+            LocalFileWriterFactory.getLocalFileWriter(
+                rssBaseConf, new File(basePath, dataFileName), dataBufferSize);
+        FileWriter indexWriter = createWriter(indexFileName, indexBufferSize)) {
```

Comment:
Why not use your new writer to write index as well?

Reply:
There are two reason.
- IndexFile is tiny which is not needed to use new writer to write.
- It is easy to serialize index structure to byte[] and write to file than serialize to a bytebuf.

Reply:
Maybe you should add some comments.